### PR TITLE
update status and deposit return msg refactor

### PR DESCRIPTION
**Pull request by janitachalam**

### CODE REVIEW

File: contracts/cw-proposal-single/src/contract.rs [hunk 1]
```diff
@@ -232,24 +232,21 @@ pub fn execute_execute(
...
     PROPOSALS.save(deps.storage, proposal_id, &prop)?;
 
-    let return_deposit = get_return_deposit_msg(&prop)?;
+    let mut refund_message = vec![];
+    match prop.deposit_info {
+        Some(deposit_info) => {
+            refund_message = get_return_deposit_msg(&deposit_info, &prop.proposer)?
+        }
+        None => {}
+    }
```
Comment (ezekiiel) [inline contracts/cw-proposal-single/src/contract.rs:249]:
```suggestion
    let refund_message = match prop.deposit_info {
        Some(deposit_info) => get_return_deposit_msg(&deposit_info, &prop.proposer)?
        None => vec![]
    }
```

File: contracts/cw-proposal-single/src/contract.rs [hunk 2]
```diff
@@ -352,18 +349,23 @@ pub fn execute_close(
...
+    let mut refund_message = vec![];
+    match &prop.deposit_info {
+        Some(deposit_info) => {
+            if deposit_info.refund_failed_proposals {
+                refund_message = get_return_deposit_msg(deposit_info, &prop.proposer)?
             }
         }
-        _ => return Err(ContractError::WrongCloseStatus {}),
+        None => {}
     }
```
Comment (ezekiiel) [inline contracts/cw-proposal-single/src/contract.rs:368]: Same as above. Can avoid the `mut`. :)

```rust
    let refund_message = match prop.deposit_info {
        Some(deposit_info) => get_return_deposit_msg(&deposit_info, &prop.proposer)?
        None => vec![]
    }
```

File: contracts/cw-proposal-single/src/state.rs [hunk 3]
```diff
@@ -128,27 +128,18 @@ pub fn get_deposit_msg(
...
-            let transfer_msg: CosmosMsg = transfer_msg.into();
-            Ok(vec![transfer_msg])
-        }
-        None => Ok(vec![]),
-    }
+pub fn get_return_deposit_msg(
+    deposit_info: &CheckedDepositInfo,
+    proposer: &Addr,
+) -> StdResult<Vec<CosmosMsg>> {
+    let transfer_msg = WasmMsg::Execute {
```
Comment (ezekiiel) [inline contracts/cw-proposal-single/src/state.rs:135]: Will error if we attempt to refund zero tokens. Just need a little check. Not your bug. :)
```suggestion
    if deposit_info.deposit.is_zero() {
        return Ok(vec![])
    }
    let transfer_msg = WasmMsg::Execute {
```

**Review by ezekiiel — COMMENTED** [review]
AMAZING! Thank you so much. Just some little nits.

File: contracts/cw-proposal-single/src/contract.rs (hunk 2 above)
Comment (janitachalam) [reply contracts/cw-proposal-single/src/contract.rs:365]: slick!

File: contracts/cw-proposal-single/src/contract.rs [hunk 4]
```diff
@@ -352,18 +346,18 @@ pub fn execute_close(
...
+    prop.update_status(&env.block);
+    if prop.status != Status::Rejected {
+        return Err(ContractError::WrongCloseStatus {});
     }
+
     let old_status = prop.status;
-    let refund_message = get_return_deposit_msg(&prop)?;
+
+    let refund_message = match &prop.deposit_info {
+        Some(deposit_info) => get_return_deposit_msg(&deposit_info, &prop.proposer)?,
```
Comment (ezekiiel) [inline contracts/cw-proposal-single/src/contract.rs:358]: Need to check that we are actually refunding failed proposals.
```suggestion
        Some(deposit_info) => if deposit_info.refund_failed_proposals {
            get_return_deposit_msg(&deposit_info, &prop.proposer)?
        } else {
            vec![]
        },
```

**Review by Callum-A — APPROVED** [review]
lgtm

**Review by ezekiiel — APPROVED** [review]

**Comment by ezekiiel** [comment]
Great work @janitachalam !

**Pull request merged**